### PR TITLE
Add Node Labels

### LINKED ISSUE #106: [AVM Module Issue]: add node_labels to default_node_pool and azurerm_kubernetes_cluster_node_pool
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### (Optional) Module Version

_No response_

### (Optional) Correlation Id

_No response_

### Description

Add vnet_subnet_id to default_node_pool and azurerm_kubernetes_cluster_node_pool

### PR DESCRIPTION
**Pull request by zioproto**
## Description
Adds Node Labels to all nodepools.

Closes #106 

## Type of Change

- [ ] Non-module change (e.g. CI/CD, documentation, etc.)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [X] Feature update backwards compatible feature updates.
  - [ ] Breaking changes.
  - [ ] Update to documentation

# Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -236,6 +238,7 @@ resource "azurerm_kubernetes_cluster_node_pool" "this" {
   enable_auto_scaling   = true
   max_count             = each.value.max_count
   min_count             = each.value.min_count
+  node_labels           = each.value.labels
```
Comment (nellyk) [inline main.tf:241]: i think you need to add this to https://github.com/Azure/terraform-azurerm-avm-ptn-aks-production/blob/6cd415a9fcac51b146fcdad20db2dbfd67784440/locals.tf#L36 as well

Comment (zioproto) [reply main.tf:241]: You are probably correct. But that list is incomplete.
Does it mean we have a bug ?

Compare the 2 lists:

https://github.com/Azure/terraform-azurerm-avm-ptn-aks-production/blob/6cd415a9fcac51b146fcdad20db2dbfd67784440/variables.tf#L102-L115

and

https://github.com/Azure/terraform-azurerm-avm-ptn-aks-production/blob/6cd415a9fcac51b146fcdad20db2dbfd67784440/locals.tf#L32-L46

Comment (zioproto) [reply main.tf:241]: I added the node labels in the locals. We will address the flattening of the other missing parameters in a different PR

File: main.tf [hunk 2]
```diff
@@ -76,6 +77,7 @@ resource "azurerm_kubernetes_cluster" "this" {
   ## Resources that only support UserAssigned
   dynamic "identity" {
     for_each = local.managed_identities.user_assigned
+
```
Comment (zioproto) [inline main.tf:80]: I think this is here because we did not merge PR #105 ?

Comment (nellyk) [reply main.tf:80]: hehehe strange

File: main.tf (hunk 1 above)
Comment (nellyk) [reply main.tf:241]: added it an issue here https://github.com/Azure/terraform-azurerm-avm-ptn-aks-production/issues/109

**Review by nellyk — APPROVED** [review]

**Pull request merged**